doc/gen_docs.go

```
@@ -0,0 +1,185 @@
+		if !child.IsAvailableCommand() || child.IsAdditionalHelpTopicCommand() {
+			continue
+		}
+		cname := name + " " + child.Name()
+		link := cname + ".md"
+		link = strings.Replace(link, " ", "_", -1)
+		buf.WriteString(fmt.Sprintf("* [%s](%s)\t - %s\n", cname, linkHandler(link), child.Short))
+
+		childLink = buf.String()
+		childrenLinks = append(childrenLinks, childLink)
+		buf.Reset()
+	}
+
+	var relatedLinks []string
+	relatedCmds := cmd.RelatedCommands()
+
+	for _, relCmd := range relatedCmds {
+		var relatedLink string
+		if !relCmd.IsAvailableCommand() || relCmd.IsAdditionalHelpTopicCommand() {
+			continue
+		}
+		rname := relCmd.CommandPath()
+		link := rname + ".md"
+		link = strings.Replace(link, " ", "_", -1)
+		buf.WriteString(fmt.Sprintf("* [%s](%s)\t - %s\n", rname, linkHandler(link), relCmd.Short))
+
+		relatedLink = buf.String()
+		relatedLinks = append(relatedLinks, relatedLink)
+		buf.Reset()
+	}
+
+	var commandLink string
+	link := name + ".md"
+	link = strings.Replace(link, " ", "_", -1)
+	buf.WriteString(fmt.Sprintf("%s", linkHandler(link)))
+	commandLink = buf.String()
+	buf.Reset()
+
+	autoGenTag := "Auto generated by spf13/cobra on " + time.Now().Format("2-Jan-2006") + "\n"
+
+	return &CmdOutline{
+		Name:          name,
+		Short:         short,
+		Long:          long,
+		UseLine:       useLine,
+		Example:       example,
+		Flags:         flagString,
+		FlagSlice:     flagSlice,
+		ParentFlags:   parentFlagString,
+		ParentLink:    parentLink,
+		ChildrenLinks: childrenLinks,
+		RelatedLinks:  relatedLinks,
+		CommandLink:   commandLink,
+		HeaderScale:   headerScale,
+		AutoGenTag:    autoGenTag,
+	}
+}
+
+func GenDocsCustomTemplate(cmd *cobra.Command, w io.Writer, linkHandler func(string) string, template *template.Template) error {
+	cmd.InitDefaultHelpCmd()
+	cmd.InitDefaultHelpFlag()
+
+	buf := new(bytes.Buffer)
+
+	cmdStruct := generateCmdOutline(cmd, linkHandler)
+
+	writeToTemplate(cmdStruct, template, buf)
```

Looks like you're swallowing the error that writeToTemplate returns